src/nvim/lua/stdlib.c

```diff
@@ -541,6 +542,31 @@ static int nlua_iconv(lua_State *lstate)
 
 #endif
 
+int nlua_read_secure(lua_State *L)
+{
+  if (lua_gettop(L) < 1) {
+    return luaL_error(L, "Expected 1 argument");
+  }
+
+  if (lua_type(L, 1) != LUA_TSTRING) {
+    luaL_argerror(L, 1, "expected string");
+  }
+
+  const char *contents = NULL;
+  size_t len = 0;
+  const char *path = lua_tolstring(L, 1, &len);
+  if (path != NULL) {
+    contents = read_secure(path);
+  }
+
+  if (contents != NULL) {
+    lua_pushstring(L, contents);
```

Comment:
I think there is a leak here and you need to free `contents`.

Also this could be optimized by additionally returning the length of a file and using `lua_pushlstring` instead, so lua doesn't have to find the null byte.

Reply:
Good catch, thanks.